src/lib/core/projection/projection.ts

```diff
@@ -0,0 +1,56 @@
+import {Injectable, Directive, ModuleWithProviders, NgModule, ElementRef} from '@angular/core';
+
+
+@Directive({
+  selector: 'md-host'
+})
+export class MdProjectionHostDirective {
+  constructor(public ref: ElementRef) {}
+}
+
+
+@Injectable()
+export class MdProjectionService {
+  private _replaceWith(toReplaceEl: HTMLElement, otherEl: HTMLElement) {
+    toReplaceEl.parentElement.replaceChild(otherEl, toReplaceEl);
+  }
+
+  /**
+   * Project an element into a host element.
+   */
+  project(ref: ElementRef, host: MdProjectionHostDirective): void {
+    const componentEl = ref.nativeElement;
+    const hostEl = host.ref.nativeElement;
+    const childNodes = componentEl.childNodes;
+    let child = childNodes[0];
+
+    // Replace ComponentEl by all its children, in place.
```

Comment:
Expand comment to something like
```ts
// We hoist all of the projected element's children out into the projected elements position
// because we *only* want to move the projected element and not its children.
```